### ISSUE TITLE
ClamScan reports max-width.gif to be infected with BC.Gif.Exploit.Agent-1425366.Agent

### ISSUE THREAD

**Issue by hermes-jr** [robert-ancell/pygif#6]
Reproduces: always  
Steps to reproduce:
- `git clone https://github.com/robert-ancell/pygif.git`
- `clamscan --infected --recursive gdk-pixbuf/`
- report would then look like:
```
pygif/test-suite/max-width.gif: BC.Gif.Exploit.Agent-1425366.Agent FOUND

----------- SCAN SUMMARY -----------
Known viruses: 6698996
Engine version: 0.98.4
Scanned directories: 19
Scanned files: 265
Infected files: 1
Data scanned: 0.88 MB
Data read: 1.05 MB (ratio 0.84:1)
Time: 38.885 sec (0 m 38 s)
```

This is most likely a false positive as [no other antivirus engine](https://www.virustotal.com/gui/file/008e2a84afed6c31b6635aa9d8c7ee2176f01a0eb0a04143883a8533d7ca33c9/details) reports this file to be infected.

I was alerted by clamscan about this file being found in gdk-pixbuf repository in macports on disk.
I asked and the maintainer of gdk-pixbuf says that [the file in question comes from the PyGIF test suite](https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/issues/150#note_752329). I just want to do a double check that the file is safe and clarify the reason of detection if possible. 

There are a few reports on the Internet which mention gdk-pixbuf and the detection in question: [1](https://gitlab.com/dave_m/clamtk/issues/78), [2](https://forums.fedoraforum.org/showthread.php?322844-Infected-files-found), [3](https://lists.opensuse.org/opensuse-factory/2020-03/msg00010.html), but none of them seem to explain why that happens. [Here](https://lists.opensuse.org/opensuse-factory/2020-03/msg00010.html) someone claims this file (detected as malicious) was intentionally added, but there's no rationale for *why* it has to be like that. I get that the image has to be wide. What I don't understand is why does it have to contain a suspicious feature.

Is it possible to investigate the issue?

I also believe it would be nice to have an explanation published for others who are also lurking. Or maybe the file can be somehow modified to stop triggering detection if it is harmless indeed. Without ruining the test suite of course.

Thanks.

**Comment by robert-ancell** [comment]
The file in question sets the width of the image to the maximum size, which could case a badly written GIF reader to have memory corruption. This is the simplest file that does this, and does nothing else and is not possible to be changed. Exploiting badly written readers is a common method of attacking software, which is why I expect clamscan has detected it as such.

This GIF file is generated in the test suite [1], so the contents can be confirmed to be as intended and not replaced by something malicious.

In conclusion, this is a false positive by clamscan with the only solution being either to ignore this detection or convince them it's not a valid case (but perhaps has been used in a previous malicious attack). Either way, hopefully the existence of pygif will ensure this type of attack wont work on any readers that use the test suite!

[1] https://github.com/robert-ancell/pygif/blob/master/make-test-suite#L376

**Comment by morfyum** [comment]
Similar issue with clamScan, thanks!

/var/lib/flatpak/repo/objects/34/xxxxxxxxxxxxxxxxxx.file: BC.Gif.Exploit.Agent-1425366.Agent FOUND
/var/lib/flatpak/runtime/org.gnome.Sdk/x86_64/3.38/xxxxxxxxxxxxxxxxxx/files/libexec/installed-tests/gdk-pixbuf/test-images/gif-test-suite/max-width.gif: BC.Gif.Exploit.Agent-1425366.Agent FOUND
/var/lib/flatpak/runtime/org.gnome.Platform/x86_64/3.38/xxxxxxxxxxxxxxxxxx/files/libexec/installed-tests/gdk-pixbuf/test-images/gif-test-suite/max-width.gif: BC.Gif.Exploit.Agent-1425366.Agent FOUND

**Comment by ballapete** [comment]
There is a second file of this kind: gdk-pixbuf-2.4x.y/tests/test-images/gif-test-suite/max-width.gif. (Probably the same, just a different path name.)
